acceptance/lib/puppet/acceptance/install_utils.rb

```diff
@@ -175,7 +175,7 @@ def install_repos_on(host, project, sha, repo_configs_dir)
               :puppet_agent_version => ENV['SUITE_VERSION'] || ENV['SHA']
             }
             # this installs puppet-agent on windows (msi), osx (dmg) and eos (swix)
-            install_puppet_agent_dev_repo_on(agent, opts)
+            install_from_build_data_url('puppet-agent', "http://builds.delivery.puppetlabs.net/puppet-agent/#{opts[:puppet_agent_sha]}/artifacts/#{opts[:puppet_agent_sha]}.yaml")
```

Comment:
My primary concern here is that we are hard-coding a private URL into the installer. I don't mind using an internal URL as a default for this method, but we should provide a way for folks external to the puppet VPN to override it with something they can use (and documenting it in the README). @kevpl is there a meaningful way for a user without access to the puppet VPN to use this method?

Reply:
Agreed

Reply:
`install_from_build_data_url` doesn't do any internal v external munging that would prevent someone from using it publicly as long as the parameter you pass to it matches what we do internally. Below I've included an explanation of how that parameter works, but I think the important part of this question is whether or not we publicly expose any builds that match this pattern at this time for an external user to try. I think the answer is no, but @melissa would know that better & be able to tell if that's a direction we're even looking to move in.

The parameter is a URL that leads to a YAML file that has a specific hash inside it:
```yaml
  :platform_data:
    <packaging_platform>:
      :artifact: "<relative URL to artifact file from YAML file>"
      :repo_config: "<relative URL to repo_config file from YAML file>"
```
(`<text>` denotes a variable above, rest of the keys are hard-coded to these values)
Beaker doesn't do anything fancy to provide the `packaging_platform` value, it needs to be provided that value per host in the hosts file (which happens by default when using beaker-hostgenerator due to [this change](https://github.com/puppetlabs/beaker-hostgenerator/pull/91/commits/3f9088cd514e8a1cfbb922bfb18c950e81c7bb2b)).

Let me know if you have any questions or comments about any of this.

Reply:
That YAML file is totally tied to our environment.

I think I'm actually kind of OK that running ci:test:aio takes a bit of hacking, since building the agent outside our walls also does. I'd expect most, if not all, external users to use ci:test:git instead of the aio version when making acceptance changes to Puppet.

Reply:
Absolutely. I think having this parameterized is the best way to go. It sounds like folks are alright with that parameter living on the project side rather than in beaker?